### PR TITLE
SPM-1700: don't block with latest schema

### PR DESCRIPTION
**Pull request by psegedy**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: database_admin/update.go [hunk 1]
```diff
@@ -99,14 +99,19 @@ func dbConn() (database.Driver, *sql.DB) {
 func UpdateDB(migrationFilesURL string) {
 	utils.ConfigureLogging()
 	migrationEnabled := utils.GetBoolEnvOrDefault("ENABLE_MIGRATION", false)
+
+	conn, db := dbConn()
+	getAdvisoryLock(db)
+	defer releaseAdvisoryLock(db)
```
Comment (michalslomczynski) [inline database_admin/update.go:105]: @psegedy
- Could you briefly sum up, why it was bad before and what it fixes after change?
- If it sleeps on select, isn't it dangerous to hold AdvisoryLock?

Comment (psegedy) [reply database_admin/update.go:105]: - it blocked deployement when you had ENABLE_MIGRATION=false but you didn't need to make any migrations
- no, it should block the new deployment

**Review by michalslomczynski — APPROVED** [review]

File: database_admin/update.go (hunk 1 above)
Comment (psegedy) [reply database_admin/update.go:105]: @michalslomczynski do you know a scenario when it could be a problem that pgAdvisoryLock is not released?

Comment (michalslomczynski) [reply database_admin/update.go]: Not on top of my head at the moment

**Review by MichaelMraka — APPROVED** [review]

**Pull request merged**